### PR TITLE
Upgrade to webflo/drupal-finder 1.3

### PR DESCRIPTION
**Pull request by webflo**
Resolves https://github.com/mglaman/phpstan-drupal/pull/757

### CODE REVIEW

**Comment by webflo** [comment]
@mglaman It should be possible to remove 'drupal_root' from the configuration completely.

**Comment by mglaman** [comment]
We should be able to test dropping the `drupal_root` param with the package's own tests, as it relies on it

**Comment by mglaman** [comment]
https://github.com/mglaman/phpstan-drupal/blob/main/tests%2Ffixtures%2Fconfig%2Fphpunit-drupal-phpstan.neon#L3

**Comment by webflo** [comment]
@mglaman Should we decleare `drupal_root` as optional in the schema? It will break many configs without it.

**Comment by mglaman** [comment]
Yeah let's define it as optional so nothing crashes

**Comment by webflo** [comment]
@mglaman I could not figure out how to make it optional in the schema.

**Comment by webflo** [comment]
I tried to use https://github.com/nette/schema/issues/27

**Comment by webflo** [comment]
Lets set it do null initialy, I don't know how to use the deprecation method because it triggers a exception right away

**Comment by mglaman** [comment]
This approach was used before https://github.com/mglaman/phpstan-drupal/blob/main/src%2FDrupal%2FDrupalAutoloader.php#L131-L134

I forgot we just added that. So manually triggering the error if the deprecated key exists

File: src/Drupal/DrupalAutoloader.php [hunk 1]
```diff
@@ -84,19 +83,21 @@ class DrupalAutoloader
...
-         * @var array{drupal_root: string, bleedingEdge: array{checkDeprecatedHooksInApiFiles: bool, checkCoreDeprecatedHooksInApiFiles: bool, checkContribDeprecatedHooksInApiFiles: bool}} $drupalParams
+         * @var array{drupal_root: string|null, bleedingEdge: array{checkDeprecatedHooksInApiFiles: bool, checkCoreDeprecatedHooksInApiFiles: bool, checkContribDeprecatedHooksInApiFiles: bool}} $drupalParams
          */
         $drupalParams = $container->getParameter('drupal');
-        $drupalRoot = realpath($drupalParams['drupal_root']);
-        $finder = new DrupalFinder();
-        $finder->locateRoot($drupalRoot);
 
+        // Trigger deprecation error if drupal_root is used.
+        if (isset($drupalParams['drupal_root'])) {
```
Comment (mglaman) [inline src/Drupal/DrupalAutoloader.php:91]: We should check if it's not `null`, since it's always set

**Comment by mglaman** [comment]
Yay!

**Pull request merged**